strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -163,11 +163,17 @@ async def handle_connection_init(self, message: ConnectionInitMessage) -> None:
         if self.connection_init_timeout_task:
             self.connection_init_timeout_task.cancel()
 
-        if message.payload is not UNSET and not isinstance(message.payload, dict):
+        payload = (
+            message.payload
+            if message.payload is not None and message.payload is not UNSET
+            else {}
+        )
+
+        if not isinstance(payload, dict):
```

Comment:
Maybe additional sanitation.  We could have a helper function for this, since string dicts are common in the protocol.
`if not isinstance(payload, dict) or not all(isinstance(key, str) for key in payload.keys())`